### ISSUE TITLE
[BONUS PAYMENT] After requesting money from a Workspace, clicking on the Workspace name doesn't do anything

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22470]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to dev on web chrome
2. Create a new workspace
3. Go to the new workspace chat (with square logo) on LHN
4. Click on + and click on request money
5. Enter amount and click on next. Now on request money page, hover over the workspace name and notice the pointer cursor is shown. If you try to click on it, nothing happens.
6. Follow the same steps on any individual user chat and notice if you click on the user on request money page, it opens the user details.

## Expected Result:
After requesting money from a Workspace, a user should be able to click on the Workspace name to be brought to the Details tab of that Workspace - example below:

![image](https://github.com/Expensify/App/assets/30815269/e20cc9a3-798a-44d3-82be-487dcef943d6)

## Actual Result:
Cursor pointer in workspace chat even when it's not clickable and does nothing when clicked

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** dev - v1.3.35-5
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/9e32ce9c-d93b-434e-b1d3-e706766d0bd1

**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688732531740269

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01005525eef2eeb6a9</li>
        <li>Upwork Job ID: 1679022727333158912</li>
        <li>Last Price Increase: 2023-07-12</li>
    </ul>
</details>

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Dev: Cursor pointer in workspace chat even when it's not clickable and does nothing when clicked
### What is the root cause of that problem?
We only open details for user profile, and do nothing for a workspace.
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/components/MoneyRequestConfirmationList.js#L212-L217
### What changes do you think we should make in order to solve the problem?
Open workspace details when the clicked item is a workspace.

```javascript
const navigateToUserDetail = (option) => {
    if (option.accountID) {
        Navigation.navigate(ROUTES.getProfileRoute(option.accountID));
        return;
    }
    if(option.reportID){
        Navigation.navigate(ROUTES.getReportDetailsRoute(option.reportID));
    }
};
```

### What alternative solutions did you explore? (Optional)

**Comment by jliexpensify** [comment]
I can't access DEV (not an Engineer). Going to reassign this to someone in Engineering to triage.

**Comment by jliexpensify** [comment]
Hi @MonilBhavsar - do you mind checking if this is a [bug](https://github.com/Expensify/App/issues/22470#issue-1794892434) on DEV? Let me know if it is and I'll handle Upworks, cheers!

**Comment by MonilBhavsar** [comment]
I am able to reproduce. I am not sure why it mentions DEV, as it is happening on staging too.
Should we open Workspace settings page here? For users, we open details page, but there is no such details page for workspace.

**Comment by jliexpensify** [comment]
> Should we open Workspace settings page here? For users, we open details page, but there is no such details page for workspace.

Hi @MonilBhavsar - that sounds like a new feature. I'm also not sure where we'd bring the user to - the Workspace itself doesn't have a "profile", so I'd think this isn't a bug.

I can ask internally though.

**Comment by jliexpensify** [comment]
@ahmedGaber93 I've updated the GH issue, would you like to amend your proposal?

**Comment by ahmedGaber93** [comment]
>I've updated the GH issue, would you like to amend your proposal?

Thanks @jliexpensify, This is exactly what my proposal try to fix.

**Comment by narefyev91** [comment]
@Pujan92 can you please describe why using this isPolicyExpenseChat is better option instead of option.policyID

**Comment by ahmedGaber93** [comment]
@narefyev91 
I think if it not account it will be workspace 100%, because we only support this two options only, as a enum type account and workspace.
and the check should be for null or undeifined values for `option.reportID` before using it
```javascript
if (option.accountID) {
       // this account
}

//else it 100% workspace
 
 // check null or undeifined option.reportID before use it.
if (option.reportID){
    Navigation.navigate(ROUTES.getReportDetailsRoute(option.reportID));
}
 ```

**Comment by Pujan92** [comment]
> @Pujan92 can you please describe why using this isPolicyExpenseChat is better option instead of option.policyID or option.reportID

@narefyev91 for all report types we don't want to show the details page. As this is specifically required for the workspace then I think better to check only for that report type which also increases the readability about only for `policyExpenseChat` we are navigating.

**Comment by ahmedGaber93** [comment]
> or all report types we don't want to show the details page. 

@Pujan92 I think there is no other reports type in request money it only account and workspace, and if there is other types here we should also support them to be clickable like accound and workspace.

Thanks!

**Comment by Pujan92** [comment]
> and if there is other types here we should also support them to be clickable like accound and workspace but it does't any other types.

@ahmedGaber93 in that case we don't require if condition which you are suggesting and directly we can make navigate call `Navigation.navigate(ROUTES.getReportDetailsRoute(option.reportID));`. I am suggesting to use specific/direct field as a check which we are allowing for better understanding/readability.

**Comment by ahmedGaber93** [comment]
@Pujan92 yes, I appreciate that 👍.
but i don't say if condition it not require, I just prefer using it for nullcheck `option.reportID`.
and may use comments in PR if i found there is something not readable.

Thanks!

**Comment by Pujan92** [comment]
@ahmedGaber93 Thanks. I just tried to put the thought which came to my mind. Anyways, I will be fine with the team's decision. :)

**Comment by narefyev91** [comment]
Proposal from @ahmedGaber93 looks good to me https://github.com/Expensify/App/issues/22470#issuecomment-1627219201
🎀 👀 🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
Sorry, I missed this. Reviewing now

**Comment by jliexpensify** [comment]
Hi @priya-jha - I notice you report a lot of issues, is there a way that we can get your hiring automated? I'm wondering why you have a few different GH profiles? Thanks!

**Comment by jliexpensify** [comment]
@priya-jha, also I'm having some issues inviting you - can you apply here? Thanks! https://www.upwork.com/jobs/~01005525eef2eeb6a9

**Comment by ahmedGaber93** [comment]
While recording videos for [my branch](https://github.com/ahmedGaber93/App/tree/issue-22470) I found hidden issue blocked this issue steps.
**When opening any workspace chat, app redirect to Concierge chat and delete workspace chat from LHN**


https://github.com/Expensify/App/assets/41129870/ce22e996-f395-4d09-9cdf-bded765dc17f

I report it in slack here
https://expensify.slack.com/archives/C049HHMV9SM/p1689681903439209

I think we need to create new issue for that, and hold the current issue util solving it.

What do you think about that? @MonilBhavsar @narefyev91

**Comment by ahmedGaber93** [comment]
I report the issue and it now in GH https://github.com/Expensify/App/issues/23126, I think we need to hold for it.

**Comment by jliexpensify** [comment]
Hi @ahmedGaber93 - just to clarify, you want to place this issue on hold and wait u til https://github.com/Expensify/App/issues/23126 is resolved?

**Comment by ahmedGaber93** [comment]
@jliexpensify yes, this new issue https://github.com/Expensify/App/issues/23126 blocked me from completing the test of this fix.

**Comment by jliexpensify** [comment]
On hold - @ahmedGaber93 give me a bump when you want this removed off hold!

**Comment by ahmedGaber93** [comment]
@jliexpensify @MonilBhavsar 
the new issue #23126 is closed as a not reproducible, and recently I found similar issue related to workspace chat saying it is a beta feature and need beta permission policyExpenseChat from backend, and the frontend permission in [contributingGuides](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#working-on-beta-features) is not enough.

I request the permission on slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1690639996486369) and waiting anyone to add me.

@MonilBhavsar can you add me or mention to anyone can add me.

**Comment by jliexpensify** [comment]
I think @MonilBhavsar can add you to that dev environment, I can't sorry!

**Comment by ahmedGaber93** [comment]
I got some help from slack, thanks!

**Comment by ahmedGaber93** [comment]
@narefyev91 [PR](https://github.com/Expensify/App/pull/23980) is now ready for review.

**Comment by situchan** [comment]
> It would be great if we could automate this process so that priya-zha profile could be tagged instead of @ Priya.

`-` is not recognized well by Melvin. I think @thienlnam is already aware of this.

**Comment by jliexpensify** [comment]
@ahmedGaber93 we can take this off hold, right?

**Comment by ahmedGaber93** [comment]
Yes @jliexpensify, it now works fine with the `policyExpenseChat` beta front and backend permission.

**Comment by jliexpensify** [comment]
Here's a Payment Summary for everyone to check:

- Bug Reporter: @priya-zha $250
- Contributor: @ahmedGaber93 $1000
- C+: @narefyev91 $1000 (Callstack - Invoice separately)

Upworks job [here](https://www.upwork.com/jobs/~01005525eef2eeb6a9).

**Comment by jliexpensify** [comment]
Bump @narefyev91 to complete the checklist!

Paying everyone else today.

**Comment by jliexpensify** [comment]
Paid and job removed. Waiting on checklist.

**Comment by MonilBhavsar** [comment]
Bumping to weekly

**Comment by narefyev91** [comment]
- The PR that introduced the bug has been identified. Link to the PR - it's not coming from any regressions - logic was missed for current behaviour
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment - no comments
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion - no discussions
- Determine if we should create a regression test for this bug - no needs any additional work here in case that it's not a regression fix

**Comment by jliexpensify** [comment]
Great, closing this out!

**Comment by situchan** [comment]
@jliexpensify sorry for late here. This GH was out of my track as I am not assigned.
But I reviewed linked [PR](https://github.com/Expensify/App/pull/23980) as I was assigned there. I think the reason for assigning different C+ on PR was @narefyev91 OOO until Aug 3 maybe.

Here's timeline analysis:
- contributor was [assigned](https://github.com/Expensify/App/issues/22470#event-9837372848) on Jul 17
- contributor [suggested](https://github.com/Expensify/App/issues/22470#issuecomment-1640105272) to put on hold on Jul 18
- issue [put](https://github.com/Expensify/App/issues/22470#event-9894404480) on hold on Jul 24 (18-24 won't affect timeline as waiting for team to confirm)
- issue seems [resumed](https://github.com/Expensify/App/issues/22470#issuecomment-1659264197) on Jul 31
- I got [assigned](https://github.com/Expensify/App/pull/23980#issuecomment-1659724797) as C+ reviewer on Aug 1
- PR was [merged](https://github.com/Expensify/App/pull/23980#event-9977957994) the same day on Aug 1

So if I am not wrong, PR was merged within 2 business days excluding holding days

**Comment by jliexpensify** [comment]
Oh interesting - so basically, we would need to pay both you and @ahmedGaber93 an extra $500 bonus each?

@MonilBhavsar can you confirm the above timeline is accurate?

**Comment by MonilBhavsar** [comment]
As mentioned in the timeline, issue was put on HOLD after contributor was assigned. So we took that hold time into consideration, and hence it was not eligible for bonus.
Though PR was merged the same day it was opened. It should be eligible for bonus if do not take into consideration the hold time

**Comment by jliexpensify** [comment]
Thanks Monil and @situchan - so if I am understanding correctly, here is what we need to pay out now:

- Contributor: @ahmedGaber93 $500 bonus
- C+: @situchan $1000 + $500 bonus

Does this mean that @narefyev91 shouldn't be paid then?

**Comment by jliexpensify** [comment]
Bump @MonilBhavsar to confirm payments :)

**Comment by MonilBhavsar** [comment]
Sorry, was ooo yesterday.
@situchan is not C+ here? @narefyev91 who is C+ is paid outside of upwork. So I think we only need to pay $500 bonus to @ahmedGaber93

**Comment by situchan** [comment]
@MonilBhavsar I reviewed https://github.com/Expensify/App/pull/23980 which linked this issue. Please correct me if I am wrong.

**Comment by MonilBhavsar** [comment]
Oh yes, Looks like Melvin assigned a reviewer other than C+ in this issue. So yes, @jliexpensify we need to pay $500 bonus to @situchan and @ahmedGaber93

**Comment by jliexpensify** [comment]
Thanks Monil - new job here: https://www.upwork.com/jobs/~0129829630e8138576

**Comment by jliexpensify** [comment]
Payments made, closing!

**Comment by situchan** [comment]
Just to confirm that I am eligible for 500 or 1500 🙂

**Comment by jliexpensify** [comment]
Oh crap, i'm sorry @situchan - I forgot you need to be paid an extra $100 as well.

**Comment by jliexpensify** [comment]
FInal payment summary:

Contributor: @ahmedGaber93 $500 bonus
C+: @situchan $1000 + $500 bonus

**Comment by situchan** [comment]
No problem. Thanks

**Comment by jliexpensify** [comment]
FINALLY SORTED. Closing again.
